### PR TITLE
tests: moved printf_float test to unittests [adopted]

### PR DESCRIPTION
**Pull request by gebart**
Since @haukepetersen will be gone for some time I decided to adopt #5384.

This is a rebase of #5384 with the comments addressed as a separate commit.

### CODE REVIEW

File: tests/unittests/tests-printf_float/tests-printf_float.c [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+#define BUFSIZE             (10)
+
+static const float in0 = 2016.0349f;
+static const float in1 = 123.4567f;
+static const float in2 = 0.0f;
+
+static void sfprintf_float(void)
+{
+    char tmp[BUFSIZE];
+    char *str = tmp;
```
Comment (A-Paul) [inline tests/unittests/tests-printf_float/tests-printf_float.c]: As in the original PR, I'm asking for the sense of the intermediate variable `tmp`. Wouldn't `char str[BUFSIZE];` serve the same purpose (and spare a small amount of memory)?

File: tests/unittests/tests-printf_float/tests-printf_float.c [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    snprintf(str, BUFSIZE, "%f", in0);
+    TEST_ASSERT_EQUAL_STRING("2016.0349", str);
+
+    snprintf(str, BUFSIZE, "%.2f", in0);
+    TEST_ASSERT_EQUAL_STRING("2016.03", str);
+
+    snprintf(str, BUFSIZE, "%.f", in0);
+    TEST_ASSERT_EQUAL_STRING("2016", str);
+
+    snprintf(str, BUFSIZE, "%.4f", in1);
```
Comment (A-Paul) [inline tests/unittests/tests-printf_float/tests-printf_float.c]: As far as you know, is this testing any other aspect as the test in line 42?

File: tests/unittests/tests-printf_float/tests-printf_float.c [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    TEST_ASSERT_EQUAL_STRING("123.4567", str);
+
+    snprintf(str, BUFSIZE, "%.1f", in1);
+    TEST_ASSERT_EQUAL_STRING("123.5", str);
+
+    snprintf(str, BUFSIZE, "%.2f", in1);
```
Comment (A-Paul) [inline tests/unittests/tests-printf_float/tests-printf_float.c]: Equivalent to test in line 51?

**Comment by gebart** [comment]
I guess that the original intent of the multiple format strings tests were to make the unit test more exhaustive with regards to rounding behaviour.

**Comment by gebart** [comment]
@A-Paul, are you satisfied with the answer that the reason for multiple calls to `sprintf` is intended to test different rounding cases?

**Comment by gebart** [comment]
ping! @A-Paul

**Comment by A-Paul** [comment]
> @A-Paul, are you satisfied with the answer that the reason for multiple calls to sprintf is intended to test different rounding cases?

Not really. :-)
Test suites are expanded by introducing more test classes or dedicated border cases. Not by repeating the same aspect.
But anyway it doesn't do any harmful, except using memory.

Please squash. ACK when Murdock is green.

**Comment by gebart** [comment]
It fails on msp430. does it make sense to cast to double before printing or will that pull in alot of extra junk?

**Comment by gebart** [comment]
Changed the test input variables from `float` to `double` to fix compilation errors on msp430.

**Comment by A-Paul** [comment]
> Changed the test input variables from `float` to `double` to fix compilation errors on msp430.

In that case, can you also remove the trailing *f's* from the values?

**Comment by gebart** [comment]
- removed `f` float specifier from variable definitions.
 - removed redundant sprintf tests
 - added sprintf test for leading spaces

**Comment by gebart** [comment]
OK to squash?

**Comment by kaspar030** [comment]
yes, please squash!

**Comment by gebart** [comment]
squashed, rebased

**Comment by kYc0o** [comment]
Murdock agrees, go.

**Pull request merged**